iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageLevelImpl.java

```diff
@@ -618,4 +619,91 @@ public void allocateVariantArrayElement(int[] indexes, int sourceIndex) {
         super.allocateVariantArrayElement(indexes, sourceIndex);
         topologyModel.allocateVariantArrayElement(indexes, sourceIndex);
     }
+
+    @Override
+    public void convertToTopology(TopologyKind newTopologyKind) {
+        Objects.requireNonNull(newTopologyKind);
+        if (newTopologyKind != topologyModel.getTopologyKind()) {
+            if (newTopologyKind == TopologyKind.NODE_BREAKER) {
+                throw new PowsyblException("Topology model conversion from bus/breaker to node/breaker not yet supported");
```

Comment:
To decrease complexity and indentations, you could add a return if the two kinds are the same, a return if the newTopologyKind is node breaker and remove the else (as the enum only has two values).

Reply:
Done, now the complexity is < 15